storage_controller/src/service.rs

```diff
@@ -2917,6 +2921,105 @@ impl Service {
         .await?
     }
 
+    pub(crate) async fn tenant_timeline_archival_config(
+        &self,
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+        req: TimelineArchivalConfigRequest,
+    ) -> Result<(), ApiError> {
+        tracing::info!(
+            "Setting archival config of timeline {tenant_id}/{timeline_id} to '{:?}'",
+            req.state
+        );
+
+        let _tenant_lock = trace_shared_lock(
+            &self.tenant_op_locks,
+            tenant_id,
+            TenantOperations::TimelineArchivalConfig,
+        )
+        .await;
+
+        self.ensure_attached_wait(tenant_id).await?;
```

Comment:
Using `tenant_remote_mutation` now

Reply:
Why is it important to use it with archival?

Reply:
It's explained here:
https://github.com/neondatabase/neon/blob/e58e045ebb80940f8fa05c8c75fdb118978fa14a/storage_controller/src/service.rs#L3088-L3091

We want all of the three things I'd say. In fact, if only some shards get archived, it would be quite bad.

Reply:
Hmm... Oki, makes sense. I guess also in the situation where only some shards would be affected we'd be making noise.